File: backend-python/app/models/contract.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import BaseApp

class Contract(BaseApp):
    __tablename__ = "contracts"

    id = Column(Integer, primary_key=True, index=True)
    tenant_id = Column(Integer, index=True)
    
    client_name = Column(String, index=True)
    client_cnpj = Column(String, index=True)
    
    start_date = Column(DateTime)
    end_date = Column(DateTime, nullable=True)
    
    total_value = Column(Float, nullable=True)
    status = Column(String, default="ACTIVE") # ACTIVE, INACTIVE
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    # Relationships
    service_orders = relationship("ServiceOrder", back_populates="contract", cascade="all, delete-orphan")

File: backend-java/src/main/java/com/nfse/saas/models/app/Contract.java
package com.nfse.saas.models.app;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "contracts")
public class Contract {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "client_name")
    private String clientName;

    @Column(name = "client_cnpj")
    private String clientCnpj;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Column(name = "total_value")
    private Double totalValue;

    private String status = "ACTIVE";

    @Column(name = "created_at", insertable = false, updatable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "contract", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<ServiceOrder> serviceOrders;

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getTenantId() { return tenantId; }
    public void setTenantId(Long tenantId) { this.tenantId = tenantId; }
    public String getClientName() { return clientName; }
    public void setClientName(String clientName) { this.clientName = clientName; }
    public String getClientCnpj() { return clientCnpj; }
    public void setClientCnpj(String clientCnpj) { this.clientCnpj = clientCnpj; }
    public LocalDateTime getStartDate() { return startDate; }
    public void setStartDate(LocalDateTime startDate) { this.startDate = startDate; }
    public LocalDateTime getEndDate() { return endDate; }
    public void setEndDate(LocalDateTime endDate) { this.endDate = endDate; }
    public Double getTotalValue() { return totalValue; }
    public void setTotalValue(Double totalValue) { this.totalValue = totalValue; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public List<ServiceOrder> getServiceOrders() { return serviceOrders; }
    public void setServiceOrders(List<ServiceOrder> serviceOrders) { this.serviceOrders = serviceOrders; }
}
